src/Infrastructure/Persistence/Repositories/RequisitionRepository.cs

```diff
@@ -27,7 +30,67 @@ public async Task<Requisition> UpdateAsync(Requisition requisition)
 
         public async Task<IReadOnlyList<Requisition>> GetAllAsync()
         {
-            return await _context.Requisitions.ToListAsync();
+            return await _context.Requisitions
+                .Include(r => r.Submitter)
+                .ToListAsync();
+        }
+
+        public async Task<PaginatedList<Requisition>> GetAllAsync(PageRequest pageRequest, DateTime? startDate, DateTime? endDate, string? expenseHead, string? department, HashSet<int>? statusFilter, HashSet<int>? typeFilter)
+        {
+            var query = _context.Requisitions.Include(r => r.Submitter).AsQueryable();
+            // Apply Search 
+            if (!string.IsNullOrWhiteSpace(pageRequest?.Keyword))
+            {
+                var keyword = pageRequest.Keyword.ToLower();
+
+                var stringProperties = typeof(Requisition).GetProperties()
+                    .Where(p => p.PropertyType == typeof(string));
+
+                foreach (var property in stringProperties)
+                {
+                    query = query.Where(r => EF.Functions.Like(EF.Property<string>(r, property.Name).ToLower(), $"%{keyword}%"));
+                }
+            }
+
+            // Additional filters based on method parameters
+            if (startDate.HasValue && startDate.Value != DateTime.MinValue)
+                query = query.Where(r => r.RequestedDate.Date >= startDate.Value.Date);
+            if (endDate.HasValue && endDate.Value != DateTime.MinValue)
+                query = query.Where(r => r.RequestedDate.Date <= endDate.Value.Date);
+            if (!string.IsNullOrWhiteSpace(expenseHead))
+                query = query.Where(r => EF.Functions.Like(r.ExpenseHead.ToLower(), expenseHead.ToLower()));
```

Comment:
why do we have `.ToLower` , db queries are mostly case-insensitive.

Reply:
Comment addressed